domain-mapping/classes/class-dm-media.php

```diff
@@ -0,0 +1,393 @@
+<?php
+/**
+ * Class DM_Media
+ *
+ * @package DarkMatter
+ *
+ * @since 2.2.0
+ */
+
+defined( 'ABSPATH' ) || die;
+
+/**
+ * Class DM_Media
+ *
+ * @since 2.2.0
+ */
+class DM_Media {
+	/**
+	 * An array of Media domains.
+	 *
+	 * @var array
+	 */
+	private $allowed_mime_types = array();
+
+	/**
+	 * An array of Media domains.
+	 *
+	 * @var array
+	 */
+	private $media_domains = array();
+
+	/**
+	 * Number of Media domains that are available.
+	 *
+	 * @var int
+	 */
+	private $media_domains_count = -1;
```

Comment:
This makes more sense as a helper method on the class, so you don't have to keep track of state that can potentially be out of date?